### PR TITLE
Clarify use of LoadFromAssemblyName method

### PR DESCRIPTION
**Pull request by ritchxu**

### CODE REVIEW

File: Documentation/design-docs/assemblyloadcontext.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ This method should be overriden in a *Custom LoadContext* if the intent is to ov
 
 ### LoadFromAssemblyName
 
-This method can be used to load an assembly into a load context different from the load context of the currently executing assembly.
+This method can be used to load an assembly into a load context different from the load context of the currently executing assembly. If invoked on *Default LoadContext*, the assembly will not be loaded into a different load context.
```
Comment (vitek-karas) [inline Documentation/design-docs/assemblyloadcontext.md]: Thanks for taking the time to improve the docs!
I think a more accurate statement would be something like this:  
> The assembly will be loaded into the load context on which the method is called. If the context can't resolve the assembly in its `Load` method the assembly loading will defer to the `Default` load context. In such case it's possible the loaded assembly is from the `Default` context even though the method was called on a non-default context.
> Calling this method directly on the `AssemblyLoadContext.Default` will only load the assembly from the `Default` context. Depending on the caller the `Default` may or may not be different from the load context of the currently executing assembly.
> This method does not "forcefully" load the assembly into the specified context. It basically initiates a bind to the specified assembly name on the specified context. That bind operation will go through the full binding resolution logic which is free to resolve the assembly from any context (in reality the most likely outcome is either the specified context or the default context). This process is described above.
> To make sure a specified assembly is loaded into the specified load context call `AssemblyLoadContext.LoadFromAssemblyPath` and specify the path to the assembly file.

Comment (ritchxu) [reply Documentation/design-docs/assemblyloadcontext.md]: Thanks @vitek-karas for elaborating! I've added your statement in the follow-up change.

File: Documentation/design-docs/assemblyloadcontext.md [hunk 2]
```diff
@@ -68,7 +68,10 @@ This method should be overriden in a *Custom LoadContext* if the intent is to ov
...
+Calling this method directly on the **AssemblyLoadContext.Default** will only load the assembly from the **Default** context. Depending on the caller the **Default** may or may not be different from the load context of the currently executing assembly.
+This method does not "forcefully" load the assembly into the specified context. It basically initiates a bind to the specified assembly name on the specified context. That bind operation will go through the full binding resolution logic which is free to resolve the assembly from any context (in reality the most likely outcome is either the specified context or the default context). This process is described above.
+To make sure a specified assembly is loaded into the specified load context call **AssemblyLoadContext.LoadFromAssemblyPath** and specify the path to the assembly file.
```
[the remark this answers is not in the document]
Comment (ritchxu) [reply Documentation/design-docs/assemblyloadcontext.md]: Nit accepted 💯

**Comment by vitek-karas** [comment]
Thanks a lot for taking the time to do this!

**Pull request merged**